wweval/R/manuscript_figs.R

```diff
@@ -0,0 +1,423 @@
+      data = hosp |> dplyr::filter(model_type == "ww"),
+      aes(
+        x = date, ymin = `0.025`, ymax = `0.975`,
+        group = as.character(forecast_date)
+      ), alpha = 0.1,
+      fill = "cornflowerblue",
+      show.legend = FALSE
+    ) +
+    geom_ribbon(
+      data = hosp |> dplyr::filter(model_type == "ww"),
+      aes(
+        x = date, ymin = `0.25`, ymax = `0.75`,
+        group = as.factor(forecast_date)
+      ),
+      alpha = 0.1,
+      fill = "cornflowerblue",
+      show.legend = FALSE
+    ) +
+    geom_line(
+      data = hosp |> dplyr::filter(model_type == "hosp"),
+      aes(
+        x = date, y = `0.5`, group = forecast_date
+      ),
+      color = "purple4"
+    ) +
+    geom_ribbon(
+      data = hosp |> dplyr::filter(model_type == "hosp"),
+      aes(
+        x = date, ymin = `0.025`, ymax = `0.975`,
+        group = as.character(forecast_date)
+      ), alpha = 0.1,
+      fill = "purple4",
+      show.legend = FALSE
+    ) +
+    geom_ribbon(
+      data = hosp |> dplyr::filter(model_type == "ww"),
+      aes(
+        x = date, ymin = `0.25`, ymax = `0.75`,
+        group = as.factor(forecast_date)
+      ),
+      fill = "purple4",
+      alpha = 0.1,
+      show.legend = FALSE
+    ) +
+    xlab("") +
+    ylab("Daily hospital admissions") +
+    ggtitle(glue::glue(
+      "{horizon_to_plot}"
+    )) +
+    theme_bw() +
+    scale_color_discrete() +
+    scale_fill_discrete() +
+    scale_x_date(
+      date_breaks = "2 weeks",
+      labels = scales::date_format("%Y-%m-%d")
+    ) +
+    theme_bw() +
+    theme(
+      axis.text.x = element_text(
+        size = 8, vjust = 1,
+        hjust = 1, angle = 45
+      ),
+      axis.title.x = element_text(size = 12),
+      axis.title.y = element_text(size = 12),
+      plot.title = element_text(
+        size = 10,
+        vjust = 0.5, hjust = 0.5
+      )
+    )
+
+  return(p)
+}
+
+#' Make CRPS underlay figure
+#'
+#' @param scores A tibble of scores by location, forecast date, date and model,
+#' the ouput of `scoringutils::score()` on samples.
+#' @param loc_to_plot A  string indicating the state abbreviations of the state
+#' to plot
+#' @param horizon_to_plot A string indicating what horizon period to plot,
+#' either `nowcast`, `1 wk`, or `4 wks`
+#'
+#' @return A ggplot object containing a bar chart of the crps score averaged
+#' across the horizon for each forecast date, colored by the model type
+#' @export
+make_crps_underlay_fig <- function(scores,
+                                   loc_to_plot,
+                                   horizon_to_plot) {
+  scores_by_horizon <- scores |>
+    dplyr::filter(location == loc_to_plot) |>
+    dplyr::mutate(
+      horizon = dplyr::case_when(
+        date <= forecast_date ~ "nowcast",
+        date > forecast_date & date <= forecast_date +
+          lubridate::days(7) ~ "1 wk",
+        date > forecast_date + lubridate::days(21) &
+          date <= forecast_date + lubridate::days(28) ~ "4 wks"
```

Comment:
For D.R.Y. reasons I think this should be broken out into its own function. Given that the results will be used for filtering later on, would be good to have a `TRUE ~ NA` at the end for "none of the above`.

A couple other questions:
1. are you sure you want string horizons instead of, say, an integer `horizon_weeks` (or for that mattter string/factor horizons that _derive_ from an integer `horizon_weeks`?)
2. For safety, would be good to specify a minimum date for the `"nowcast"` if possible.

Reply:
Good call, I also repeat this a bunch of times. Will make these changes tomorrow! 

1. I think that makes the most sense, bc in Fig 3 we have one that is continuous, and this also makes it easier to set the order properly. The only thing is that the nowcast is 10 days so this will be -2 and -1 but the -2 will be incomplete. I think its fine though since we will mostly be analyzing the outputs using this categorically. 
2. Agreed-- though as this will apply to both scores and quantiles we might just have to do something where you pass in the maximum number of days of the nowcast instead of using the data (since the calibration data will be present in the quantiles, but not in the scores)